coordinator/ethereum/message-anchoring-i2/src/test/kotlin/linea/anchoring/MessageAnchoringAppTest.kt

```diff
@@ -0,0 +1,243 @@
+    ).asList(),
+    createL1MessageSentV1Logs(
+      blockNumber = 200UL,
+      contractAddress = L1_CONTRACT_ADDRESS,
+      messageNumber = 3UL,
+      messageHash = "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa03".decodeHex(),
+      rollingHash = "0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb03".decodeHex()
+    ).asList(),
+    createL1MessageSentV1Logs(
+      blockNumber = 400UL,
+      contractAddress = L1_CONTRACT_ADDRESS,
+      messageNumber = 4UL,
+      messageHash = "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa04".decodeHex(),
+      rollingHash = "0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb04".decodeHex()
+    ).asList()
+  )
+  private lateinit var vertx: Vertx
+
+  @BeforeEach
+  fun setUp(vertx: Vertx) {
+    this.vertx = vertx
+    l2MessageService = FakeL2MessageService(contractAddress = L2_CONTRACT_ADDRESS)
+    l1Client = FakeEthApiClient(
+      initialLogsDb = emptySet(),
+      topicsTranslation = mapOf(
+        L1RollingHashUpdatedEvent.topic to "L1RollingHashUpdatedEvent",
+        MessageSentEvent.topic to "MessageSentEvent",
+        L2RollingHashUpdatedEvent.topic to "L2RollingHashUpdatedEvent"
+      ),
+      log = LogManager.getLogger("FakeEthApiClient")
+    )
+
+    configureLoggers(
+      rootLevel = Level.INFO,
+      "FakeEthApiClient" to Level.INFO
+    )
+  }
+
+  private fun createApp(
+    anchoringTickInterval: Duration = 100.milliseconds,
+    l1PollingInterval: Duration = 100.milliseconds,
+    l1EventSearchBlockChunk: UInt = 1000u,
+    l1EventPollingTimeout: Duration = 2.seconds,
+    l1SuccessBackoffDelay: Duration = 1.milliseconds,
+    messageQueueCapacity: UInt = 100u,
+    maxMessagesToAnchorPerL2Transaction: UInt = 10u
+  ): MessageAnchoringApp {
+    return MessageAnchoringApp(
+      vertx = vertx,
+      config = MessageAnchoringApp.Config(
+        l1PollingInterval = l1PollingInterval,
+        l1SuccessBackoffDelay = l1SuccessBackoffDelay,
+        l1ContractAddress = L1_CONTRACT_ADDRESS,
+        l2HighestBlockTag = BlockParameter.Tag.LATEST,
+        anchoringTickInterval = anchoringTickInterval,
+        l1RequestRetryConfig = RetryConfig.noRetries,
+        l1EventPollingTimeout = l1EventPollingTimeout,
+        l1EventSearchBlockChunk = l1EventSearchBlockChunk,
+        messageQueueCapacity = messageQueueCapacity,
+        maxMessagesToAnchorPerL2Transaction = maxMessagesToAnchorPerL2Transaction
+      ),
+      l1EthApiClient = l1Client,
+      l2MessageService = l2MessageService
+    )
+  }
+
+  @Test
+  fun `should anchor messages from fresh deployment`() {
+    l1Client.setLogs(l1MessageSentLogs.flatten())
+
+    val anchoringApp = createApp(anchoringTickInterval = 100.milliseconds, l1EventSearchBlockChunk = 100u)
+    anchoringApp.start().get()
+    l1Client.setLatestBlockTag(l1MessageSentLogs.last().last().blockNumber + 10UL)
```

Comment:
imo, the latest block tag should be better set higher than the finalized block tag